translator/translate/otel/pipeline/prometheus/translator.go

```diff
@@ -73,7 +74,9 @@ func (t *translator) Translate(conf *confmap.Conf) (*common.ComponentTranslators
 			),
 			Exporters: common.NewTranslatorMap(awsemf.NewTranslatorWithName(common.PipelineNamePrometheus)),
 			Extensions: common.NewTranslatorMap(agenthealth.NewTranslator(agenthealth.LogsName, []string{agenthealth.OperationPutLogEvents}),
-				agenthealth.NewTranslatorWithStatusCode(agenthealth.StatusCodeName, nil, true)),
+				agenthealth.NewTranslatorWithStatusCode(agenthealth.StatusCodeName, nil, true),
+					ecsobserver.NewTranslator(),
```

Comment:
We should only add the ecsobserver translator if the ecsservicediscovery section is configured.